README.md

```diff
@@ -12,144 +12,361 @@
+
+#### Kafka
+```go
+go get github.com/luno/workflow/adapters/kafkastreamer
+```
+
+#### Reflex
+```go
+go get github.com/luno/workflow/adapters/reflexstreamer
+```
+
+#### SQL Store
+```go
+go get github.com/luno/workflow/adapters/sqlstore
+```
+
+#### SQL Timeout
+```go
+go get github.com/luno/workflow/adapters/sqltimeout
+```
+---
+## Usage
+
+### Step 1: Define the workflow
+```go
+package usage
+
+import (
+	"context"
+
+	"github.com/luno/workflow"
+)
+
+type Step int
+
+func (s Step) String() string {
+	switch s {
+	case StepOne:
+		return "One"
+	case StepTwo:
+		return "Two"
+	case StepThree:
+		return "Three"
+	default:
+		return "Unknown"
+	}
+}
+
+const (
+	StepUnknown Step = 0
+	StepOne Step = 1
+	StepTwo Step = 2
+	StepThree Step = 3
+)
+
+type MyType struct {
+	Field string
+}
+
+func Workflow() *workflow.Workflow[MyType, Step] {
+	b := workflow.NewBuilder[MyType, Step]("my workflow name")
+
+	b.AddStep(StepOne, func(ctx context.Context, r *workflow.Record[MyType, Step]) (Step, error) {
+		r.Object.Field = "Hello,"
+		return StepTwo, nil
+	}, StepTwo)
+
+	b.AddStep(StepTwo, func(ctx context.Context, r *workflow.Record[MyType, Step]) (Step, error) {
+		r.Object.Field += " world!"
+		return StepThree, nil
+	}, StepThree)
+
+	return b.Build(...)
+}
+```
+```mermaid
+---
+title: The above defined workflow creates the below Directed Acyclic Graph
+---
+stateDiagram-v2
+	direction LR
+    
+	[*]-->One
+    One-->Two
+    Two-->Three
+    Three-->[*]
+```
+### Step 2: Run the workflow
+```go
+wf := usage.Workflow()
+
+ctx := context.Background()
+wf.Run(ctx)
+```
+**Stop:** To stop all processes and wait for them to shut down correctly call
+```go
+wf.Stop()
+```
+### Step 3: Trigger the workflow
+```go
+foreignID := "82347982374982374"
+runID, err := wf.Trigger(ctx, foreignID, StepOne)
+if err != nil {
+	...
+}
+```
+**Awaiting results:** If appropriate and desired you can wait for the workflow to complete. Using context timeout (cancellation) is advised.
+```go
+foreignID := "82347982374982374"
+runID, err := wf.Trigger(ctx, foreignID, StepOne)
+if err != nil {
+	...
+}
+
+ctx, cancel := context.WithTimeout(ctx, 10 * time.Second)
+defer cancel()
+
+record, err := wf.Await(ctx, foreignID, runID, StepThree)
+if err != nil {
+	...
+}
+```
+
+### Detailed examples
+Head on over to `./example` to get familiar with **callbacks**, **timeouts**, **testing**, **connectors** and
+ more about the syntax in depth 😊
+
+---
+
+## Workflow's RunState
+RunState is the state of a workflow run and can only exist in one state at any given time. RunState is a
+ finite state machine and allows for control over the workflow run. A workflow run is every instance of
+ a triggered workflow.
+```mermaid
+---
+title: Diagram the run states of a workflow
+---
+stateDiagram-v2
+	direction LR
+    
+	[*]-->Initiated
+
+    Initiated-->Running
+    
+    Running-->Completed
+    Running-->Paused
+
+    Paused-->Running
```

Comment:
Can we label the run states here? I wasn't sure what it was, had to look in the mermaid code 👍